src/SonarScanner.MSBuild.Common/ProcessRunnerArguments.cs

```diff
@@ -266,4 +216,75 @@ private static string ShellEscape(string argLine)
         }
         return sb.ToString();
     }
+
+    public record Argument(string Value, bool Escaped = false)
```

Comment:
This looks good to me as a solution👍.
Consider:
```suggestion
    public readonly record struct Argument(string Value, bool Escaped = false)
```